src/Test.Utilities.Sarif/TestRuleBehaviors.cs

```diff
@@ -41,6 +41,7 @@ public enum TestRuleBehaviors
         RegardOptionsAsInvalid = 0x4000,
 
         RaiseExceptionProcessingBaseline = 0x8000,
-        RaiseExceptionPostingLogFile = 0x10000
+        RaiseExceptionPostingLogFile = 0x10000,
+        RaiseStackOverflowException = 0x20000
```

Comment:
>RaiseStackOverflowException [](http://example.com/codeflow?start=8&length=27)

You have mixed three things in this PR, stack overflow creation, interface/loggings changes, a new trace, and fixed a functional bug in the analysis engine.

Can you go back and just implement the test behaviors change? I will provide a clean example of tracing for now.